File: example/kotlin/src/testFixtures/kotlin/com/some/pkg/othermodule/fixtures/Asserts.kt
package com.some.pkg.othermodule.fixtures

import org.assertj.core.api.Assertions.assertThat

import com.some.pkg.othermodule.api.*

data class ExpectedOtherProperty(
    var id: String? = null,
    var name: String? = null,
)
fun assertOtherProperty(given: OtherProperty, expectedInit: ExpectedOtherProperty.() -> Unit) {
    val expected = ExpectedOtherProperty().apply(expectedInit)

    expected.id?.let {
        assertThat(given.getId().value).isEqualTo(it)
    }

    expected.name?.let {
        assertThat(given.name).isEqualTo(it)
    }
}

data class ExpectedOtherClass(
    var id: String? = null,
    var amount: Int? = null,
)
fun assertOtherClass(given: OtherClass, expectedInit: ExpectedOtherClass.() -> Unit) {
    val expected = ExpectedOtherClass().apply(expectedInit)

    expected.id?.let {
        assertThat(given.id.value).isEqualTo(it)
    }

    expected.amount?.let {
        assertThat(given.amount).isEqualTo(it)
    }
}

File: example/typescript/test/OtherModule/Fixtures/Asserts.ts
namespace OtherModule.Assert {
    export interface ExpectedOtherProperty {
        id?: string,
        name?: string,
    }
    export function otherProperty(given: OtherProperty, expected: ExpectedOtherProperty) {
        if (expected.id !== undefined) {
            AssertEquals(given.getId().value, expected.id)
        }

        if (expected.name !== undefined) {
            AssertEquals(given.name, expected.name)
        }
    }

    export interface ExpectedOtherClass {
        id?: string,
        amount?: number,
    }
    export function otherClass(given: OtherClass, expected: ExpectedOtherClass) {
        if (expected.id !== undefined) {
            AssertEquals(given.id.value, expected.id)
        }

        if (expected.amount !== undefined) {
            AssertEquals(given.amount, expected.amount)
        }
    }
}